src/libraries/TickMath.sol

```diff
@@ -1,19 +1,24 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 pragma solidity ^0.8.0;
 
+import {CustomRevert} from "./CustomRevert.sol";
+
 /// @title Math library for computing sqrt prices from ticks and vice versa
 /// @notice Computes sqrt price for ticks of size 1.0001, i.e. sqrt(1.0001^tick) as fixed point Q64.96 numbers. Supports
 /// prices between 2**-128 and 2**128
 library TickMath {
+    using CustomRevert for bytes4;
+
     /// @notice Thrown when the tick passed to #getSqrtPriceAtTick is not between MIN_TICK and MAX_TICK
-    error InvalidTick();
+    error InvalidTick(int24 tick);
     /// @notice Thrown when the price passed to #getTickAtSqrtPrice does not correspond to a price between MIN_TICK and MAX_TICK
     error InvalidSqrtPrice();
 
     /// @dev The minimum tick that may be passed to #getSqrtPriceAtTick computed from log base 1.0001 of 2**-128
     int24 internal constant MIN_TICK = -887272;
     /// @dev The maximum tick that may be passed to #getSqrtPriceAtTick computed from log base 1.0001 of 2**128
     int24 internal constant MAX_TICK = 887272;
+    uint256 internal constant MAX_TICK_UINT256 = 887272;
```

Comment:
why can't you just cast MAX_TICK instead of writing another constant?

Reply:
thought it would increase gas but it didnt. Have reverted that